src/components/BoxGrid.tsx

```diff
@@ -3,6 +3,35 @@ import styled from "styled-components"
 
 import Emoji from "./Emoji"
 
+export interface IOpener {
+  isOpen: boolean
+}
+
+export interface IGridStyle {
+  columnNumber: number
+}
+
+export interface IColor {
+  color: string
+}
+
+export interface IBaseItem {
+  emoji: string
+  title: string
+  description: string
+}
+
+export interface IBoxStyle extends IOpener, IGridStyle, IColor {}
+
+export interface IGridItemProps extends IBaseItem, IBoxStyle {
+  index: number
+  callback: (idx: number) => void
+}
+
+export interface IBoxProps {
```

Comment:
As we are only exporting 1 component in this file, rename this interface to `IProps` to keep it consistent with the rest of the components.

I would suggest only exporting this interface and `IBaseItem` as they are the only interfaces the consumers are going to interact with. I would keep the rest of the interfaces private (no `export`).

Reply:
I have rename `IBaseItem` into `IBoxItem` and followed your recommendations